airbyte-scheduler/src/main/java/io/airbyte/scheduler/temporal/TemporalClient.java

```diff
@@ -38,12 +43,31 @@ public TemporalClient(WorkflowClient client) {
     this.client = client;
   }
 
-  public ConnectorSpecification submitGetSpec(long jobId, int attempt, JobGetSpecConfig config) {
-    final IntegrationLauncherConfig integrationLauncherConfig = new IntegrationLauncherConfig()
+  public OutputAndStatus<JobOutput> submitGetSpec(long jobId, int attempt, JobGetSpecConfig config) {
+    final IntegrationLauncherConfig launcherConfig = new IntegrationLauncherConfig()
         .withJobId(jobId)
         .withAttemptId((long) attempt)
         .withDockerImage(config.getDockerImage());
-    return getWorkflowStub(SpecWorkflow.class, TemporalJobType.GET_SPEC).run(integrationLauncherConfig);
+    return toOutputAndStatus(() -> getWorkflowStub(SpecWorkflow.class, TemporalJobType.GET_SPEC).run(launcherConfig));
+
+  }
+
+  public OutputAndStatus<JobOutput> submitCheckConnection(long jobId, int attempt, JobCheckConnectionConfig config) {
+    final StandardCheckConnectionInput input = new StandardCheckConnectionInput().withConnectionConfiguration(config.getConnectionConfiguration());
+    final IntegrationLauncherConfig launcherConfig = new IntegrationLauncherConfig()
+        .withJobId(jobId)
+        .withAttemptId((long) attempt)
+        .withDockerImage(config.getDockerImage());
+
+    return toOutputAndStatus(() -> getWorkflowStub(CheckConnectionWorkflow.class, TemporalJobType.CHECK_CONNECTION).run(launcherConfig, input));
+  }
+
+  private OutputAndStatus<JobOutput> toOutputAndStatus(CheckedSupplier<JobOutput, TemporalJobException> supplier) {
+    try {
+      return new OutputAndStatus<>(JobStatus.SUCCEEDED, supplier.get());
+    } catch (TemporalJobException e) {
```

Comment:
Since you're also throwing runtime, what do you expect the behavior for the output ?

Reply:
yup. the exception handling is not good right now. i will be looking at it across the board today and fix these.